packages/dds/tree/src/test/feature-libraries/editable-tree/editableTree.edit.spec.ts

```diff
@@ -0,0 +1,199 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { fail, strict as assert } from "assert";
+import { fieldSchema, InMemoryStoredSchemaRepository, SchemaData } from "../../../schema-stored";
+import { JsonableTree, EmptyKey, rootFieldKey } from "../../../tree";
+import { ISharedTree } from "../../../shared-tree";
+import { brand } from "../../../util";
+import {
+    singleTextCursor, getTypeSymbol, isEmptyTree, insertRootSymbol, insertNodeSymbol, appendNodeSymbol,
+    isEditableFieldSequence, isUnwrappedNode, FieldKinds,
+} from "../../../feature-libraries";
+import { ITestTreeProvider, TestTreeProvider } from "../../utils";
+import {
+    addressSchema, complexPhoneSchema, ComplexPhoneType, emptyNode, expectTreeSequence, fullSchemaData, Int32,
+    int32Schema, optionalChildSchema, personData, PersonType, phonesSchema, PhonesType, schemaMap, stringSchema,
+} from "./mocks";
+
+async function createSharedTrees(schemaData: SchemaData, data?: JsonableTree, nofTrees = 1):
+    Promise<readonly [ITestTreeProvider, readonly ISharedTree[]]> {
+    const provider = await TestTreeProvider.create(nofTrees);
+    for (const tree of provider.trees) {
```

Comment:
If you use the schema editing approach suggested below, you should only need to do it on one of the trees since it will be sent to the others as an op.